### ISSUE TITLE
VPN Workflow Problem

### ISSUE THREAD

**Issue by four43** [kontena/kontena#150]
Ubuntu seems to be a target platform for Kontena, which is great. I'm using Ubuntu as a my dev/workstation and Ubuntu servers as master and agents. When setting up VPN I ran into some issues. I got the .ovpn file, went to import it using the gnome-network-manager but it doesn't recognize the inline keys. This isn't really your problem, more of a limitation of the network manager, but if you could output a tar file with the certs separate, it might ease adoption a bit. Just a thought!

Bug Tracker for Ubuntu: https://bugs.launchpad.net/ubuntu/+source/network-manager-openvpn/+bug/606365

Thanks for all the work on Kontena, looking forward to using it once I get it setup.

**Comment by four43** [comment]
I'm playing with this now, might have a little cli command you could add to the docs to maybe get setup easier/faster.

**Comment by four43** [comment]
Still having some issues:

```bash
~$ sudo openvpn --config ./kontena-vpn/konetna-vpn.ovpn 
Thu Sep  3 14:12:04 2015 OpenVPN 2.3.2 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [EPOLL] [PKCS11] [eurephia] [MH] [IPv6] built on Dec  1 2014
Thu Sep  3 14:12:04 2015 Control Channel Authentication: tls-auth using INLINE static key file
Thu Sep  3 14:12:04 2015 UDPv4 link local: [undef]
Thu Sep  3 14:12:04 2015 UDPv4 link remote: [AF_INET]{remote_ip}:1194
Thu Sep  3 14:12:05 2015 [54.210.193.104] Peer Connection Initiated with [AF_INET]{remote_ip}:1194
Thu Sep  3 14:12:08 2015 TUN/TAP device tun0 opened
Thu Sep  3 14:12:08 2015 do_ifconfig, tt->ipv6=0, tt->did_ifconfig_ipv6_setup=0
Thu Sep  3 14:12:08 2015 /sbin/ip link set dev tun0 up mtu 1500
Thu Sep  3 14:12:08 2015 /sbin/ip addr add dev tun0 local 192.168.255.6 peer 192.168.255.5
RTNETLINK answers: Invalid argument
Thu Sep  3 14:12:08 2015 ERROR: Linux route add command failed: external program exited with error status: 2
Thu Sep  3 14:12:08 2015 Initialization Sequence Completed

{Hangs, ^C pressed}

Thu Sep  3 14:21:59 2015 event_wait : Interrupted system call (code=4)
Thu Sep  3 14:21:59 2015 /sbin/ip addr del dev tun0 local 192.168.255.10 peer 192.168.255.9
Thu Sep  3 14:21:59 2015 SIGINT[hard,] received, process exiting

```

I thought it was strange it was looking to tunnel 192.168.x.x because the network I am on is a 10.* network. However that could just be my lack of VPN knowledge. Any thoughts?

**Comment by four43** [comment]
IP Forwarding seems to be enabled....
```bash
~$ sysctl net.ipv4.ip_forward
net.ipv4.ip_forward = 1
```

**Comment by jakolehm** [comment]
What ubuntu version are you using (workstation)?

**Comment by four43** [comment]
14.04LTS on all machines I am working with. Thanks!

**Comment by jakolehm** [comment]
192.168.255.0/24 network is used only with vpn client/server.  Could you run openvpn command with more verbose output, like this:
```
~$ sudo openvpn --verb 6 --config ./kontena-vpn/konetna-vpn.ovpn 
```

And report what it says just before `RTNETLINK answers: Invalid argument` .

**Comment by jakolehm** [comment]
Ok, just tested this with default Ubuntu 14.04 virtual machine and it seems that docker bridge reserves the same ip address on workstation as inside the nodes. Easiest workaround for this is to change your workstation docker bridge address to something else than 172.17.0.0/16.

Edit `/etc/default/docker` and add `--bip 10.200.42.1/16` (or some subnet that does not create conflict). Then reboot your workstation.

In addition to this you need to tweak ovpn file a bit:
-  change `route 172.17.42.1 255.255.255.254` to `route 172.17.42.1 255.255.255.255`
  - (this is a bug, I'll add separate issue about it)
- add resolvconf scripts to end of file:
```
up /etc/openvpn/update-resolv-conf
down /etc/openvpn/update-resolv-conf
```

After these changes you must start openvpn with `--script-security 2` flag:

```
~$ sudo openvpn --script-security 2 --config ./kontena-vpn/konetna-vpn.ovpn
```

**Comment by four43** [comment]
Perfect, I'll give that a shot, then we can maybe tweak the docs a bit for
the Ubuntu install to make this go a little cleaner for other folks. Thanks
for the ideas. I'll report back.

-Seth

**Comment by jakolehm** [comment]
@four43 did those instructions solve your problem?

**Comment by four43** [comment]
Sorry @jakolehm, we were moving offices and networks, I wanted to make sure this was still an issue for me and that it wasn't caused by anything else. Now that we are all settled in, I can look at this again.

1. I am still able to duplicate the issue I was having earlier on our new network setup.
2. Output of OpenVPN, using the `--verb 6` option, as you probably saw:
```
...
Wed Sep  9 10:52:56 2015 us=542991 PUSH: Received control message: 'PUSH_REPLY,dhcp-option DNS 8.8.4.4,dhcp-option DNS 8.8.8.8,route 192.168.255.1,topology net30,ping 10,ping-restart 60,ifconfig 192.168.255.6 192.168.255.5'
Wed Sep  9 10:52:56 2015 us=543073 OPTIONS IMPORT: timers and/or timeouts modified
Wed Sep  9 10:52:56 2015 us=543093 OPTIONS IMPORT: --ifconfig/up options modified
Wed Sep  9 10:52:56 2015 us=543108 OPTIONS IMPORT: route options modified
Wed Sep  9 10:52:56 2015 us=543122 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
Wed Sep  9 10:52:56 2015 us=543360 ROUTE_GATEWAY 192.168.11.1/255.255.255.0 IFACE=wlan0 HWADDR=c4:e9:84:1d:9f:d0
Wed Sep  9 10:52:56 2015 us=543937 TUN/TAP device tun0 opened
Wed Sep  9 10:52:56 2015 us=543972 TUN/TAP TX queue length set to 100
Wed Sep  9 10:52:56 2015 us=544001 do_ifconfig, tt->ipv6=0, tt->did_ifconfig_ipv6_setup=0
Wed Sep  9 10:52:56 2015 us=544036 /sbin/ip link set dev tun0 up mtu 1500
Wed Sep  9 10:52:56 2015 us=545779 /sbin/ip addr add dev tun0 local 192.168.255.6 peer 192.168.255.5
Wed Sep  9 10:52:56 2015 us=548897 /sbin/ip route add 10.81.0.0/16 via 192.168.255.5
Wed Sep  9 10:52:56 2015 us=552937 /sbin/ip route add 172.17.42.1/31 via 192.168.255.5
RTNETLINK answers: Invalid argument
```
3. Added `--bip 10.200.42.1/16` to the Docker conf, I am on a 192.168.x.x network now.
4. Restart Docker `sudo service docker restart`
5. Updated the VPN config route map from `255.255.255.254` to `255.255.255.255`, added resolvers at the bottom.

That worked! Excellent fix! Now to make that a little less scary for others. Sounds like you have a patch up. Need help updating the docs perhaps?

Thanks for the help and looking into this promptly.

-Seth

**Comment by jakolehm** [comment]
OpenVPN image is already patched so the route problem should be fixed. Just shoot a pull request if you have improvements to docs, any help is appreciated :)

**Comment by four43** [comment]
Still having some issues here. VPN is connected, it says "Initialization Sequence Completed" so I left a terminal window open with that running yet. However I'm still having issues resolving "registry.kontena.local" from within a script. I then tried it in my browser and got a `ERR_NAME_NOT_RESOLVED` error. If I run dig I get this:

```
~$ dig registry.kontena.local

; <<>> DiG 9.9.5-3ubuntu0.2-Ubuntu <<>> registry.kontena.local
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 59257
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;registry.kontena.local.		IN	A

;; ANSWER SECTION:
registry.kontena.local.	1	IN	A	10.81.21.50

;; Query time: 38 msec
;; SERVER: 172.17.42.1#53(172.17.42.1)
;; WHEN: Wed Sep 09 15:23:29 CDT 2015
;; MSG SIZE  rcvd: 78
```

`docker push` is still giving me fits though: `dial tcp: lookup registry.kontena.local: no such host`

Thoughts?

**Comment by four43** [comment]
For reference: 
```
~$ curl http://registry.kontena.local/v1/
curl: (6) Could not resolve host: registry.kontena.local
```

**Comment by jakolehm** [comment]
Did you add following lines to .ovpn file?
```
up /etc/openvpn/update-resolv-conf
down /etc/openvpn/update-resolv-conf
```
(without those ubuntu does not seem to update it's dns settings)

**Comment by four43** [comment]
I did, it added the entries to resolv.conf:

``` bash
# /etc/resolv.conf
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
nameserver 172.17.42.1
nameserver 8.8.4.4
nameserver 8.8.8.8
search kontena.local mydomain.com
```

Like I said, dig works, but others seem to fail, which is super odd.

**Comment by four43** [comment]
As expected running `host` against that resolv.conf:

```bash
~$ host registry.kontena.local
registry.kontena.local has address 10.81.21.50
Host registry.kontena.local not found: 3(NXDOMAIN)
Host registry.kontena.local not found: 3(NXDOMAIN)
```

**Comment by four43** [comment]
Talking with a friend of mine, he recommended checking out /etc/nsswitch.conf if things are resolving weird:

```bash
# /etc/nsswitch.conf
...
hosts:          files mdns4_minimal [NOTFOUND=return] dns mdns4
networks:       files
...
```

By default, glibc programs that do DNS queries will just use an mdns probe on the network and let whoever respond it seems.

I found this Avahi-Daemon that is part of Ubuntu by default. It is a zeroconf networking solution. It tries to listen for `*.local` address and resolve them via mDNS from what I can tell: http://ubuntuforums.org/showthread.php?t=1339516

Something something about .local unicast dns servers....

**Comment by jakolehm** [comment]
Interesting, what happens if you just try to resolve `registry` name (without kontena.local)?

**Comment by four43** [comment]
Ah! That is working! Assuming because it is one of the search domains?

**Comment by jakolehm** [comment]
Yes, search domain causes that.. I think `registry.<grid_name>` should also work (might be handy if you have multiple grids)

**Comment by four43** [comment]
ehhhh kind of:
```
~$ host registry.mygrid.kontena.local
registry.mygrid.kontena.local has address 10.81.21.50
Host registry.mygrid.kontena.local not found: 3(NXDOMAIN)
Host registry.mygrid.kontena.local not found: 3(NXDOMAIN)
~$ host registry.mygrid
Host registry.mygrid not found: 3(NXDOMAIN)
~$ host registry.mygrid.
Host registry.mygrid. not found: 3(NXDOMAIN)
$ host registry.mygrid.kontena.local
registry.mygrid.kontena.local has address 10.81.21.50
Host registry.mygrid.kontena.local not found: 3(NXDOMAIN)
Host registry.mygrid.kontena.local not found: 3(NXDOMAIN)
~$ host registry.mygrid
Host registry.mygrid not found: 3(NXDOMAIN)
~$ host registry
registry.kontena.local has address 10.81.21.50
Host registry.kontena.local not found: 3(NXDOMAIN)
Host registry.kontena.local not found: 3(NXDOMAIN)
```

**Comment by four43** [comment]
So, now the new problem. If I do: `docker push registry/namespace/container-name 1.0.0` it tries to push to the docker.io (their hub) lol. SO CLOSE

**Comment by jakolehm** [comment]
I have to admin that I didn't expect so many problems when using Ubuntu desktop (longtime user and usually linux has fewer problems vs win/osx when it comes to network related stuff). As a temporary workaround you can always push images using ip address (ip will change only when you update/deploy registry). Just remember to use dns name on service definition when pointing to internal images (so that they work even if ip changes).

**Comment by jakolehm** [comment]
@nevalla can you confirm if these problems exist on newer Ubuntu releases?

**Comment by four43** [comment]
I can push to `registry.` (note the dot) to make sure docker knows it's a domain name, not just a namespace. This seems to work. It's ugly, but it works.

**Comment by four43** [comment]
Things that seem to work well:

1. Changes mentioned above to VPN config, sounds like those are on un-released code already, sweet.
2. Changes made to binding IP of docker, I used `--bip 192.168.254.1/24`
3. Turning off Ubuntu's dumb avahi-daemon, not needed if your network is setup right.
```bash
sudo service avahi-daemon stop;
sudo sh -c "echo manual > /etc/init/avahi-daemon.override";
```
(via http://askubuntu.com/questions/205937/how-can-i-disable-avahi-daemon)

**Comment by four43** [comment]
I think we could get around this if we didn't use `.local`. Thoughts on switching to just `.kontena`? `registry.kontena` for example. Possibly look into `registry.mygrid.kontena` if the registry is per grid?

**Comment by four43** [comment]
Now I'm having issues connecting to the VPN again after the weekend.

`kontena vpn config` outputs: `{"error":"Not found"}`

Yet `kontena vpn create` outputs `Vpn already exists`

:(

**Comment by four43** [comment]
The VPN client is trying to connect to one of the grid members, not that master. Is that normal? I forget if it did that before it broke...

**Comment by jakolehm** [comment]
Don't know the details but this sounds like a docker upgrade regression (1.6 -> 1.7 may loose reference to volume). Vpn should recover if you re-deploy it ( `kontena service deploy vpn` ).

**Comment by jakolehm** [comment]
Btw you can see vpn service details with following commands:
```
$ kontena service show vpn
$ kontena service logs vpn
```

Maybe it helps with debugging.

**Comment by four43** [comment]
Thanks for the tips, I think I found the issue.

We are running our Kontena cluster on AWS using their Security Groups to power our firewall. I verified the IP the OpenVPN client is trying to connect to is the correct IP.

```
~$ kontena service show vpn
mygrid/vpn:
  status: running
  stateful: yes
  scaling: 1
  image: kontena/openvpn:ethwe
  dns: vpn.mygrid.kontena.local
  affinity: 
    - node==ip-10-0-0-99
  cmd: 
  env: 
    - OVPN_SERVER_URL=udp://{{OUR-VPN-IP}}:1194
  ports:
    - 1194:1194/udp
  volumes:
  volumes_from:
  links: 
  cap_add:
    - NET_ADMIN
  cap_drop:
  containers:
```

Up and running, looks good, right?

I ran `openvpn --verb 6 --script-security 2 --config ./kontena-vpn/konetna-vpn.ovpn` and found that the UDP packets don't seem to be making it to the server:

` UDPv4 WRITE [42] to [AF_INET]{{OUR-VPN-IP}}:1194: P_CONTROL_HARD_RESET_CLIENT_V2 kid=0 pid=[ #1 ] [ ] pid=0 DATA len=0`

It does that for a while, I tried telnet (TCP) too:

```
~$ telnet {{OUR-VPN-IP}} 1194
Trying {{OUR-VPN-IP}}...
telnet: Unable to connect to remote host: Connection refused
```

So what the heck! I logged into the grid server, the one hosting the VPN service and found this:

```
~$ sudo netstat -tulpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1155/sshd       
tcp        0      0 127.0.0.1:8080          0.0.0.0:*               LISTEN      5133/docker-proxy
tcp6       0      0 :::22                   :::*                    LISTEN      1155/sshd       
udp        0      0 0.0.0.0:43769           0.0.0.0:*                           646/dhclient    
udp        0      0 0.0.0.0:68              0.0.0.0:*                           646/dhclient    
udp6       0      0 :::1194                 :::*                                1395/docker-proxy
udp6       0      0 :::42286                :::*                                646/dhclient  
```
Port 1194 is only bound on an IPv6 address. I'm using IPv4 style addresses to connect. Could this be causing my issue?

Thanks for all the help guys, we'll get it!

**Comment by nevalla** [comment]
Have you opened port 1194 on Security Group's inbound rules?

**Comment by four43** [comment]
I have for TCP and UDP. From what I can tell AWS Security groups only deal with IPv4 addresses at this time.

**Comment by four43** [comment]
Perhaps more pressing, I'm getting: `kontena vpn config` outputting `{"error":"Not found"}` still, even after the re-deploy. showing the VPN service is same as above, looks to be running.

**Comment by nevalla** [comment]
How about `$ kontena service logs vpn`? Are there any hints what's going on?

**Comment by four43** [comment]
There are no entries since last Friday, the 11th, when everything was working. It's like the VPN service became totally dissociated from the rest of Kontena.

**Comment by nevalla** [comment]
Is it the same result if you drop vpn service (`$kontena vpn delete`) and then re-create it?

**Comment by four43** [comment]
Haven't tried that:

```
~$ kontena vpn delete
{"code":503,"message":"Node unavailable","backtrace":null}
```

Interesting. 
```
~$ kontena grid list
Name                           Nodes    Services     Users     
mygrid *                       1        0            1    
```

**Comment by nevalla** [comment]
how about `$ kontena node list`?

**Comment by four43** [comment]
I updated and upgraded the servers to make sure I am on the latest version, still having issues.
```
~$ kontena node list
Name                           OS                                       Driver          Labels                         Status    
ip-10-0-0-99                   Ubuntu 14.04.2 LTS (3.13.0-48-generic)   aufs            -                              online  
```
That looks to be the correct node, the VPN service's affinity seems to point to that one.

Thanks for the help @nevalla , appreciating the input.

**Comment by four43** [comment]
So I can stop and start the vpn container but I can't delete it.

```
~$ kontena service delete vpn = nope
~$ kontena service stop vpn = service stopped when I check it
~$ kontena service start vpn = service started again when I check it
```

**Comment by four43** [comment]
Also checking the logs through kontena cli only shows log entries from last Friday, however I can check logs on the kontena node directly through docker and there are entries from Sunday, slightly better, still not today however.

**Comment by nevalla** [comment]
can you also check logs from kontena-agent container. Seems that there might be something wrong.

**Comment by four43** [comment]
On the Kontena Node: 
```
I, [2015-09-15T16:40:00.399835 #1]  INFO -- QueueWorker: initialized
I, [2015-09-15T16:40:00.400846 #1]  INFO -- WebsocketClient: initialized with token Ln5to3nH8M+HOS3kIE1yKNoWV4Us085iCK05HCzrirXxTRRHY6z4swueguyxowsqIe2F5/v8trLAW/n5RtgcnQ==
I, [2015-09-15T16:40:00.401523 #1]  INFO -- LogWorker: initialized
I, [2015-09-15T16:40:00.402175 #1]  INFO -- WeaveAttacher: initialized
I, [2015-09-15T16:40:00.403291 #1]  INFO -- CadvisorLauncher: initialized
I, [2015-09-15T16:40:00.403334 #1]  INFO -- StatsWorker: initialized
I, [2015-09-15T16:40:00.403376 #1]  INFO -- EtcdLauncher: initialized
I, [2015-09-15T16:40:00.414118 #1]  INFO -- WeaveAttacher: fetching containers information
I, [2015-09-15T16:40:00.532226 #1]  INFO -- NodeInfoWorker: publishing node information
I, [2015-09-15T16:40:00.532361 #1]  INFO -- ContainerInfoWorker: fetching containers information
I, [2015-09-15T16:40:00.532765 #1]  INFO -- EventWorker: started to stream docker events
I, [2015-09-15T16:40:00.536490 #1]  INFO -- StatsWorker: waiting for cadvisor
I, [2015-09-15T16:40:00.543213 #1]  INFO -- LogWorker: starting to stream logs for container: 6857ac291cbbe7ccd36b564e1b406f32a0847966f6b516b88b9a0b958bab90d1
I, [2015-09-15T16:40:00.552073 #1]  INFO -- StatsWorker: cadvisor is running, starting stats loop
E, [2015-09-15T16:40:00.892273 #1] ERROR -- ContainerInfoWorker: publish_info: no such id: 6d9ec81b2b414661aea36b03f0c36b44ba8fb3b66ff05b0078104307321a47b8

E, [2015-09-15T16:40:00.965903 #1] ERROR -- ContainerInfoWorker: publish_info: no such id: 6d9ec81b2b414661aea36b03f0c36b44ba8fb3b66ff05b0078104307321a47b8

I, [2015-09-15T16:40:02.919854 #1]  INFO -- LogWorker: starting to stream logs for container: 91b64349330787e9e95943af6f04709ca357503b1d43938ccbf7798692f38bf9
I, [2015-09-15T16:40:20.554579 #1]  INFO -- WebsocketClient: connection closed with code: 1006
I, [2015-09-15T16:40:41.660170 #1]  INFO -- WebsocketClient: connection closed with code: 1006




I, [2015-09-15T16:41:00.979588 #1]  INFO -- ContainerInfoWorker: fetching containers information
I, [2015-09-15T16:41:00.981648 #1]  INFO -- WebsocketClient: connection closed with code: 1006
/usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/socket.rb:139:in `rescue in block in connect': connect timeout reached (Excon::Errors::Timeout)
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/socket.rb:119:in `block in connect'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/socket.rb:110:in `each'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/socket.rb:110:in `connect'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/socket.rb:28:in `initialize'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/connection.rb:389:in `new'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/connection.rb:389:in `socket'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/connection.rb:106:in `request_call'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/middlewares/mock.rb:47:in `request_call'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/middlewares/instrumentor.rb:22:in `request_call'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/middlewares/base.rb:15:in `request_call'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/middlewares/base.rb:15:in `request_call'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/middlewares/base.rb:15:in `request_call'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/middlewares/base.rb:15:in `request_call'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon/connection.rb:233:in `request'
	from /usr/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib/excon.rb:229:in `get'
	from /app/lib/kontena/helpers/node_helper.rb:8:in `node_info'
	from /app/lib/kontena/weave_adapter.rb:111:in `start!'
	from /app/lib/kontena/agent.rb:31:in `start!'
	from /app/bin/kontena-agent:27:in `<main>'
```

It doesn't seem to be connecting to the master?

I think we're on to something! The master seems happy though...

**Comment by four43** [comment]
I GOT IT! The two Kontena servers couldn't talk to each other. Somewhere they got blocked on TCP port 8080. I destroyed the VPN and re-created it, downloaded the new config and it's working!

**Comment by four43** [comment]
That said, I think there is maybe an underlying issue with stale data and not being able to connect to the Kontena Nodes... All indications from the kontena-cli pointed to the node being okay, `kontena node list` made it seem fine.

**Comment by four43** [comment]
I'm gonna open a separate issue for that, as it probably doesn't just pertain to VPN, then we can get back on topic, make some docs for Ubuntu specific business then call it good. Whew. Thanks @nevalla

**Comment by nevalla** [comment]
Great to hear you got it working! We definitely need to check why Kontena master thought node was online. We have already pushed some effort to this issue #95, but apparently more investigation is needed.

**Comment by four43** [comment]
Thanks @nevalla, appreciate the help as always.

**Comment by jakolehm** [comment]
Closing this because vpn should work after these:
https://github.com/kontena/kontena/pull/265
https://github.com/kontena/kontena/pull/267
